deduped-dashmap/src/lib.rs

```diff
@@ -264,9 +289,86 @@ pub enum ControlFlow<C> {
+    use std::collections::{HashMap, HashSet};
+
+    #[test]
+    fn test_simple() {
+        // Set up a map with some values related to days of the week
+        let map = DedupedMap::<&str, usize, &str>::new();
+        map.insert("monday", 0, "week day");
+        map.insert("tuesday", 1, "week day");
+        map.insert("wednesday", 2, "week day");
+        map.insert("thursday", 3, "week day");
+        map.insert("friday", 4, "week day");
+        map.insert("saturday", 5, "week end");
+        map.insert("sunday", 6, "week end");
+
+        // Make some assertions about the initial state
+        assert_eq!(map.len_pointers(), 7);
+        assert_eq!(map.len_storage(), 2);
+
+        let mut pointers = map.pointers.clone().into_iter().collect::<Vec<_>>();
+        pointers.sort_by_key(|v| v.0);
+        let storage = map.storage.clone().into_iter().collect::<HashMap<_, _>>();
+
+        let cases = vec![
+            ("friday", 4, "week day", 5),
+            ("monday", 0, "week day", 5),
+            ("saturday", 5, "week end", 2),
+            ("sunday", 6, "week end", 2),
+            ("thursday", 3, "week day", 5),
+            ("tuesday", 1, "week day", 5),
+            ("wednesday", 2, "week day", 5),
+        ];
+        for (idx, (day, day_num, day_type, refcount)) in cases.into_iter().enumerate() {
+            assert_eq!(pointers[idx].0, day);
+            assert_eq!(pointers[idx].1.meta, day_num);
+            assert_eq!(
+                storage[&pointers[idx].1.hash],
+                MapValue {
+                    value: day_type,
+                    refcount,
+                }
+            );
+        }
+
+        // Remove some items from the map
+        map.remove("sunday");
+        assert_eq!(map.len_pointers(), 6);
+        assert_eq!(map.len_storage(), 2);
+        map.remove("saturday");
+        assert_eq!(map.len_pointers(), 5);
+        assert_eq!(map.len_storage(), 1);
+        map.remove("monday");
+        assert_eq!(map.len_pointers(), 4);
+        assert_eq!(map.len_storage(), 1);
+
+        // Make sure the internal structure changed as expected
+        let mut pointers = map.pointers.clone().into_iter().collect::<Vec<_>>();
+        pointers.sort_by_key(|v| v.0);
+        let storage = map.storage.clone().into_iter().collect::<HashMap<_, _>>();
+
+        let cases = vec![
+            ("friday", 4, "week day", 4),
+            ("thursday", 3, "week day", 4),
+            ("tuesday", 1, "week day", 4),
+            ("wednesday", 2, "week day", 4),
+        ];
+        for (idx, (day, day_num, day_type, refcount)) in cases.into_iter().enumerate() {
+            assert_eq!(pointers[idx].0, day);
+            assert_eq!(pointers[idx].1.meta, day_num);
+            assert_eq!(
+                storage[&pointers[idx].1.hash],
+                MapValue {
+                    value: day_type,
+                    refcount,
+                }
+            );
+        }
```

Comment:
Nit: can you also remove the rest of the items and check the invariant for the sake of completeness?

Reply:
Is the change I made what you meant?

Reply:
Yep, looks good to me, thanks!